tests/test_derivatives.py

```diff
@@ -421,7 +421,7 @@ def test_fd_adjoint(self, so, ndim, derivative, adjoint_name):
         deriv = getattr(f, derivative)
         coeff = 1 if derivative == 'dx2' else -1
         expected = coeff * getattr(f, derivative).evaluate.subs({x.spacing: -x.spacing})
-        assert deriv.T.evaluate == expected
+        assert simplify(deriv.T.evaluate) == simplify(expected)
```

Comment:
Why is simplify now needed?

Reply:
Because now subs is ours fully here and creates `(1/_h_x) * f(x)` instead of `f(x)/h_x` so doesn match without simplify (compiler refactorize so no flop increase issue)

Reply:
I could make it evaluate but that's be quite expensive for more complex expressions and not necessary.